eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -55,7 +55,8 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: A
         channel ! INPUT_RESTORED(state)
         FinalChannelId(state.channelId) -> channel
       }.toMap
-      goto(DISCONNECTED) using DisconnectedData(previousKnownAddress, channels, MAX_RECONNECT_INTERVAL) // when we restart, we will attempt to reconnect right away, but then we'll wait
+      val firstReconnectionMillis = (MAX_RECONNECT_INTERVAL.toMillis / 2) + Random.nextInt((MAX_RECONNECT_INTERVAL.toMillis / 2).toInt) // between 50% and 100% of MAX_RECONNECT_INTERVAL
```

Comment:
Why not type `firstReconnectionMillis` right here? Resolving to milliseconds seems a bit overkill.

Alternatively:

```scala
val firstNextReconnectionDelay: FiniteDuration = MAX_RECONNECT_INTERVAL.minus(Random.nextInt(MAX_RECONNECT_INTERVAL.toSeconds.toInt / 2).seconds)
```

Reply:
Yes it's a bit overkill considering we're picking a value between 30min and 1h